remote/users/src/app/users/users.service.ts

```diff
@@ -38,15 +38,26 @@ export class UsersService {
   }
 
   async create(user: User): Promise<User>  {
```

Comment:
Would using the same async flavor make sense to avoid mixing `async|await` with _thenables_?
```suggestion
  async create(user: User): Promise<User> {
    return this.usersRepository.save({
      ...user,
      password: await bcrypt.hash(user.password, 10),
    });
  }
```

Reply:
Hi! 

In this case, it's not possible; `bcrypt.hash` returns `void` no matter what, so it is necessary to wrap it in a promise to extract the result of the callback.

Reply:
I may be missing something. It looks to me like the `bcrypt` library supports promises as well.
> [with promises](https://www.npmjs.com/package/bcrypt#with-promises)
> bcrypt uses whatever Promise implementation is available in global.Promise. NodeJS >= 0.12 has a native Promise implementation built in. However, this should work in any Promises/A+ compliant implementation.